collagraph/collagraph.py

```diff
@@ -24,9 +25,31 @@
 VIRTUAL_NODE_TYPES = {"template", "slot"}
 
 
+class NotifyChangeWatcher(Watcher):
+    """
+    Custom watcher that simply calls callback function without evaluating
+    the new value of the watched expression.
+    This makes it possible to just signal to Collagraph that a dependency
+    has changed and then Collagraph can decide when to actually evaluate
+    the new value.
+    """
+
+    def update(self):
+        """On update, simply run the callback."""
+        self.callback()
+
+
+def watch(fn, cb, lazy=False, deep=True, **kwargs):
```

Comment:
Do you need deep=True? I would think it's going to watch exactly as specifically as it needs too.

Reply:
But... if the render function depends on that nested array, it should trigger, right?

Reply:
OK, I see the test, where you are rendering to a dict. But does it matter for a real renderer?

Reply:
So if I use a starred expression to really get all the values from the nested array, then it works with a shallow watcher:
```python
def render(self):
    return h("Counter", {"prop": [*self.props["prop"]]})
```
But I think having a shallow watcher might result in subtle reactivity problems, so my preference would be to keep the deep watchers for now.

Reply:
That's fine, it's your choice.

Just know that it comes at the cost of performance and is shallow in Vue.js. I suppose it's because you would never just render a full datastructure in the browser.